File: openmp_part_I/omp_daxpy-template.f90
program omp_daxpy

   integer,parameter :: N=1e7
   real*8,save  :: x(N),y(N),z(N)
   real*8 :: a = 5./3.

   call tick(it0,it1)
   call daxpy(N,a,x,y,z)
   call tock(it0,it1)

end program

subroutine daxpy(N,a,x,y,z)
   integer :: N
   real*8  :: a,x(N),y(N),z(N)

!your OMP directive goes here. Plesae use appropriate shared() term

!another OMP here
   do i = 1,N
      x(i) = real(i)*real(i)
      y(i) = (i+1.)*(i-1.)
   end do

!another OMP
   do i = 1,N
      z(i) = x(i)*a+y(i)
   end do
!And OMP END goes here

end subroutine

File: openmp-wg-2016/omp_daxpy-template.c
// daxpy.c
//
// Compute a daxpy serially.
//
//

#include "ticktock.h"

void daxpy(int n, double a, double* x, double* y, double* z) 
{
    //your pragma goes here. make sure you have correct shared(...) term
    {
    int i = 0;
    //another pragma goes here
    for (i=0; i<n; i++) {
        x[i] = (double)i*(double)i;
        y[i] = (i+1.)*(i-1.);
    }
    //the last pragma goes here
    for (i=0; i<n; i++) 
        z[i] += a * x[i] + y[i];
    }
}

int main() 
{
    int     n = 1e7;
    double* x = malloc(sizeof(double) * n);
    double* y = malloc(sizeof(double) * n);
    double* z = malloc(sizeof(double) * n);
	double  a = 5./3.;
    tick_tock tt;
    tick(&tt);
    daxpy(n,a,x,y,z);
    tock(&tt);
    free(x);
    free(y);
    free(z);
}
